desktop/main-app/src/utils/URLProtocol.ts

```diff
@@ -3,18 +3,39 @@ import runtime from "./Runtime";
 import closeAPP from "./CloseAPP";
 import { windowManager } from "./WindowManager";
 
-const actionHandler = {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    active: (_arg: URLSearchParams) => {
+class ActionHandler {
```

Comment:
class should not be concerned with specific logic

```typescript
import { app } from "electron";
import runtime from "./Runtime";
import closeAPP from "./CloseAPP";
import { windowManager } from "./WindowManager";

class URLProtocolHandler {
    public readonly handlers: ActionHandler;

    public constructor(handlers: ActionHandler) {
        this.handlers = Object.freeze(handlers);
    }

    public execute(url: string): void {
        const actionInfo = URLProtocolHandler.getActionInfo(url);

        if (actionInfo) {
            if (URLProtocolHandler.focus()) {
                this.handlers[actionInfo.name](actionInfo.args);
            }
        }
    }

    private static focus(): boolean {
        const mainWindow = windowManager.getMainWindow()?.window;
        if (mainWindow) {
            if (mainWindow.isMinimized()) {
                mainWindow.restore();
            }
            mainWindow.focus();
        }

        return !!mainWindow;
    }

    private static getActionInfo(url: string): ActionInfo {
        try {
            const data = new URL(url);
            const actionName = data.hostname as ActionNames;

            if (!protocol.handlers[actionName]) {
                return null;
            }

            return {
                name: actionName,
                args: data.searchParams,
            };
        } catch (_err) {
            return null;
        }
    }
}

const protocol = new URLProtocolHandler({
    active: () => {
        // nothing...
    },
});

const mac = () => {
    app.on("open-url", (event, url) => {
        event.preventDefault();

        protocol.execute(url);
    });
};

const win = () => {
    // in any case, this must be called, otherwise second-instance will not be triggered
    const lock = app.requestSingleInstanceLock();

    const url = process.argv.slice().pop()!;

    // act of opening by url protocol should not launch a new app (except for the first open)
    if (url.startsWith("x-agora-flat-client://") && !lock) {
        return closeAPP();
    }

    protocol.execute(url);

    app.on("second-instance", (_event, command) => {
        protocol.execute(command.pop()!);
    });
};

export const URLProtocol = runtime.isMac ? mac : win;

type ActionNames = "active";
type ActionHandler = {
    [key in ActionNames]: (arg: URLSearchParams) => void;
};
type ActionInfo = {
    name: ActionNames;
    args: URLSearchParams;
} | null;
```